packages/catalog-process/src/authMiddleware.ts

```diff
@@ -1,47 +1,112 @@
+/* eslint-disable functional/immutable-data */
+/* eslint-disable @typescript-eslint/naming-convention */
 import { ZodiosRouterContextRequestHandler } from "@zodios/express";
+import { logger } from "pagopa-interop-commons";
 import { match, P } from "ts-pattern";
+import { z } from "zod";
+import { AuthData } from "../../commons/src/auth/authData.js";
+import { readAuthDataFromJwtToken } from "../../commons/src/auth/jwt.js";
 import { ExpressContext } from "./app.js";
-import { readClaimsFromJwtToken } from "./auth/jwt.js";
-import { CatalogProcessError, ErrorTypes } from "./model/domain/errors.js";
+import {
+  CatalogProcessError,
+  ErrorTypes,
+  missingHeader,
+} from "./model/domain/errors.js";
 import { ApiError, makeApiError } from "./model/types.js";
 
+const ipRegex = /^(?:[0-9]{1,3}\.){3}[0-9]{1,3}$/;
+
+const Headers = z.object({
+  Authorization: z.string(),
+  "X-Correlation-Id": z.string(),
+  "X-Forwarded-For": z.string().ip().optional(),
+});
+
+type Headers = z.infer<typeof Headers>;
+
 export const authMiddleware: ZodiosRouterContextRequestHandler<
   ExpressContext
 > = (req, res, next) => {
+  const addCtxAuthData = (headers: Headers): void => {
+    const authorizationHeader = headers.Authorization.split(" ");
+    if (
+      authorizationHeader.length !== 2 ||
+      authorizationHeader[0] !== "Bearer"
+    ) {
+      logger.warn(
+        `No authentication has been provided for this call ${req.method} ${req.url}`
+      );
+      throw new CatalogProcessError(
+        "Authorization Illegal header key.",
+        ErrorTypes.MissingBearer
+      );
+    }
+
+    const jwtToken = authorizationHeader[1];
+    const authData = readAuthDataFromJwtToken(jwtToken);
+
+    match(authData)
+      .with(
+        P.shape({
```

Comment:
I don't think `P.shape` is needed to match objects (it's used if you want to call further methods on an object pattern)

Reply:
removed, now using otherwise for success case and match Error instance to throw CatalogProcessError